### ISSUE TITLE
[cygnus-common][CygnusApplication][YAFS] Thread not alive after agent configuration was modified.

### ISSUE THREAD

**Issue by pmo-sdr** [telefonicaid/fiware-cygnus#1624]
YAFS detects not alive threads while cygnus loads new agent configuration.

It seems like this behaviour only appears when Cygnus was started through script, If runned as a service no error occurs.

Script example:
```
{FLUME_HOME}/bin/cygnus-flume-ng agent --conf ${FLUME_HOME}/conf -f ${FLUME_HOME}/conf/agent.conf -n cygnusagent -Duser.timezone=UTC -Dfile.encoding=UTF-8
```

Error thrown:
```
time=2019-04-01T12:08:02.653Z | lvl=ERROR | corr=N/A | trans=N/A | srv=N/A | subsrv=N/A | comp=cygnusagent | op=run | msg=com.telefonica.iot.cygnus.nodes.CygnusApplication$YAFS[474] : Thread found not alive, exiting Cygnus. ID=17, name=SinkRunner-PollingRunner-DefaultSinkProcessor
```

Looks like the problem is YAFS looking for old configuration thread instead for new ones, or looking for them before they started.

**Comment by pmo-sdr** [comment]
Commited our solution proposal (5de52a2), it looks to be working but a deeper test is needed. If it succeed we will do a new PR

**Comment by fgalan** [comment]
I'd suggest to put the fix in a PR. That way is easeir for us do some kind of testing (e2e meainly)

Is this issue related with https://github.com/telefonicaid/fiware-cygnus/issues/1616? Looking to the log trace the message seems to be similar...

**Comment by pmo-sdr** [comment]
> 
> 
> Is this issue related with #1616? Looking to the log trace the message seems to be similar...

Both cause YAFS to exit Cygnus, but for different causes, so they also have different fixes.

- #1616 -> Mismatching Jetty's thread names. It could be related to the Flume update.
- #1624 -> YAFS was failing to detect configuration hot changes, causing Cygnus exit. It's an old issue on our "script-launched" Cygnus.

**Comment by fgalan** [comment]
Thanks for the clarification!

**Comment by fgalan** [comment]
Fixed by PR #1625
